gammapy/modeling/fit.py

```diff
@@ -895,3 +899,82 @@ def _repr_html_(self):
             return self.to_html()
         except AttributeError:
             return f"<pre>{html.escape(str(self))}</pre>"
+
+
+class FitResults(collections.abc.Sequence):
+    def __init__(self, results, axis=None):
+        if axis and axis.nbin != len(results):
```

Comment:
I dont know if we should have an axis here

Reply:
Maybe instead of an attribute of the class, it could be an input in relevant methods? (e.g. table creation)